### ISSUE TITLE
[TypeScript] Inadequate Type Definitions for OneSignal.init parameters

### ISSUE THREAD

**Issue by PrinceGupta1999** [OneSignal/react-onesignal#124]
Hi!

I was recently integrating this library into my app and found some of the type definitions are very broad for the `OneSignal.init` function parameters. For ex:  `IInitObject.welcomeNotification` and has a type of object so I had to reference the parameters from the [documentation](https://documentation.onesignal.com/docs/web-sdk#init-welcomenotification-parameters). Similarly `notifiyButton`, `webhooks` etc. are also defined as objects.

Defining and exporting these definitions and possibly adding the documentations would make integrations easier and less error prone

**Comment by DanyPell** [comment]
Yep. Same for promptOptions. Has a type of `object` so had to do something digging to figure out that we can have `actionMessage`, `acceptButton`, and `acceptCancel` in there. Although, adding promptOptions is buggy because it shows the prompt right upon initing instead of when we trigger a prompt.

**Comment by sherwinski** [comment]
Hey everyone and apologies for the silence on this. I'm happy to look into this more to see how we can improve type definition support. Granted I won't be able to get to it immediately but it is on our team's radar.

Linking to #108 as it's related

**Comment by fadi-george** [comment]
Should see better type interfaces with the release of 3.2.0
